### PR TITLE
feat(*): add generic logging interface

### PR DESCRIPTION
**Pull request by adamreese**
Adds a generic logging interface to importable packages.

### CODE REVIEW

File: pkg/logger/logger.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package logger
+
+// Logger provides a generic way of handeling logging.
```
Comment (thomastaylor312) [inline pkg/logger/logger.go]: Nit: spelling -> handling

**Review by thomastaylor312 — APPROVED** [review]
One small nit that could be fixed with the rebase. Otherwise this is good to go.

**Comment by johnw188** [comment]
Question - when I was building out #2342 I was assuming that there would be a concept of source (and of less importance, level). The idea was to allow the helm client to make a log request filtered to a given source and level - hook logs, pod logs, test logs, system logs, error logs, etc.

This would allow the `helm install` command to subscribe to `hook` logs in order to print kube output from hook execution, a `helm logs [release]` command to print logs from all kube objects created by a release, while other commands could subscribe to `error` logs to print detailed failure information.

Would it be possible to add the concept of a source/level to this generic interface? I don't really want to introduce a second log interface for client bound logs, I feel it would be confusing. We could have PrintF be a default system log for brevity and have a second function to specify source/level if needed.

**Comment by thomastaylor312** [comment]
@johnw188: So @adamreese and I had a big discussion about this offline. I also wanted to have leveled logs and use something like `logrus` to get some other nice features. The problem is that other projects import and use the helm code and we didn't want to have to force our choice of logger on anyone that has helm as a dependency. So for sake of making it clean and easy, we went with this. Does that make sense? I think that these logs are completely separate from the work you are doing because they are only with logs from Tiller itself. 

If you have some ideas about how to keep it clean and introduce levels, I'd love to see an example. @adamreese please chime in if I missed anything

**Comment by thomastaylor312** [comment]
@adamreese I checked your small fixes, and they look good, so feel free to merge if you are ready

**Review by technosophos — APPROVED** [review]
I like the feel of the function instead of the full interface.

**Pull request merged**